client/my-sites/themes/theme-upload/index.jsx

```diff
@@ -58,8 +57,7 @@ class Upload extends React.Component {
 		installing: React.PropTypes.bool,
 	};
 
-	constructor( props ) {
-		super( props );
+	componentDidMount() {
```

Comment:
There was no issue with `clearThemeUpload` being called at the right time (we want to do it whenever the component is mounted, as long as there is no upload in progress), but it seemed that setting state in the constructor was causing the React warning mentioned in the PR description.

Moving the call to `componentDidMount()` gives us the same behaviour without the warning.